src/System.IO.UnmanagedMemoryStream/tests/UmsCtorTests.cs

```diff
@@ -34,18 +34,114 @@ public static unsafe void PointerCtor()
+                Assert.Throws<ArgumentOutOfRangeException>(() => stream.Initialize(pByte, 1, -4, FileAccess.Read));
+                Assert.Throws<ArgumentOutOfRangeException>(() => stream.Initialize(pByte, 5, 4, FileAccess.Read));
+                Assert.Throws<ArgumentOutOfRangeException>(() => stream.Initialize(pByte, 1, 4, (FileAccess)12345));
+                stream.Initialize(pByte, 1, 4, FileAccess.ReadWrite);
+                Assert.Throws<InvalidOperationException>(() => stream.Initialize(pByte, 1, 4, FileAccess.ReadWrite));
+
+                Assert.True(stream.CanRead);
+                Assert.True(stream.CanSeek);
+                Assert.True(stream.CanWrite);
+
+                Assert.Equal(1, stream.Length);
+                Assert.Equal(4, stream.Capacity);
+                Assert.Equal(0, stream.Position);
+
+                Assert.Throws<InvalidOperationException>(() => stream.ReadTimeout);
+                Assert.Throws<InvalidOperationException>(() => stream.ReadTimeout = 42);
+                Assert.Throws<InvalidOperationException>(() => stream.WriteTimeout);
+                Assert.Throws<InvalidOperationException>(() => stream.WriteTimeout = 42);
+
+                Assert.True(stream.PositionPointer == pByte);
+            }
+        }
+
+        [Fact]
+        public static unsafe void BufferCtor()
+        {
+            const int length = 99;
+            using (FakeSafeBuffer buffer = new FakeSafeBuffer(length))
+            using (var stream = new DerivedUnmanagedMemoryStream(buffer, length, FileAccess.Read))
+            {
+                Assert.True(stream.CanRead);
+                Assert.True(stream.CanSeek);
+                Assert.False(stream.CanWrite);
+
+                Assert.Equal(length, stream.Length);
+                Assert.Equal(length, stream.Capacity);
+                Assert.Equal(0, stream.Position);
             }
+
+            using (FakeSafeBuffer buffer = new FakeSafeBuffer(length))
+            using (var stream = new DerivedUnmanagedMemoryStream())
+            {
+                Assert.False(stream.CanRead);
+                Assert.False(stream.CanSeek);
+                Assert.False(stream.CanWrite);
+
+                stream.Initialize(buffer, 0, length, FileAccess.Write);
+                Assert.Throws<InvalidOperationException>(() => stream.Initialize(buffer, 0, length, FileAccess.Write));
+
+                Assert.False(stream.CanRead);
+                Assert.True(stream.CanSeek);
+                Assert.True(stream.CanWrite);
+
+                Assert.Equal(length, stream.Length);
+                Assert.Equal(length, stream.Capacity);
+                Assert.Equal(0, stream.Position);
+
+                Assert.Throws<NotSupportedException>(() => stream.SetLength(1));
+            }
+        }
+    }
+
+    internal sealed unsafe class DerivedUnmanagedMemoryStream : UnmanagedMemoryStream
```

Comment:
Sure, will add; it was the only way to get at the protected constructors and to be able to test the state of the object before it's initialized, testing attempting to initialize it twice, etc.